OpenRA.Mods.Common/Traits/Player/EnemyWatcher.cs

```diff
@@ -85,11 +84,28 @@ public void Tick(Actor self)
 				if (lastKnownActorIds.Contains(actor.Actor.ActorID))
 					continue;
 
+				var notificationPlayed = playedNotifications.Contains(actor.Trait.Info.Notification);
```

Comment:
I think this can be moved into `AnnounceOnSeen`, doesn't it? You could then lose the bool argument to `OnDiscovered`.